test/framework/src/main/java/org/elasticsearch/cluster/ESAllocationTestCase.java

```diff
@@ -98,6 +99,10 @@ public OptionalDouble getForecastedWriteLoad(IndexMetadata indexMetadata) {
         public void refreshLicense() {}
     };
 
+    public static final DesiredBalanceShardsAllocator.ShardAllocationExplainer DUMMY_EXPLAINER = (
```

Comment:
naming nit: maybe `TEST_EXPLAINER` or `TEST_ONLY_EXPLAINER` or something like that?